pyperf/_utils.py

```diff
@@ -317,9 +320,34 @@ def __exit__(self, *args):
 class ReadPipe(_Pipe):
     def open_text(self):
         file = open(self._fd, "r", encoding="utf8")
+        if self._timeout:
+            os.set_blocking(file.fileno(), False)
         self._file = file
         return file
 
+    def read_text(self):
+        with self.open_text() as rfile:
+            if self._timeout:
+              return self._read_text_timeout(rfile, self._timeout)
+            else:
+                return rfile.read()
+
+    def _read_text_timeout(self, rfile, timeout):
+        start_time = time.time()
+        while True:
+            if time.time() - start_time > timeout:
+                raise TimeoutError(f"Timed out after {timeout} seconds")
+            ready, _, _ = select.select([rfile], [], [], timeout)
+            if ready:
+                data = rfile.read()
+                if data:
+                    return data
```

Comment:
You should read a few characters (ex: 1024) and put it in a list, rather than blocking on `.read()`, no?

Once done (once data is empty), `return ''.join(output)`.

Reply:
The data that we read from these pipes are not massive so it should not be a big problem but I agree with you, let's break it down in smaller chunks.

Reply:
select() doesn't tell you how many bytes are available. Only that at least one byte is available.

Reply:
I've implemented, have a look!